### PR TITLE
NIFI-4800 Expose the flattenMode as property in FlattenJSON processor

### PR DESCRIPTION
**Pull request by yjhyjhyjh0**
Expose all 3 kinds of flatten mode.
Documentation along with unit tests.
Finish local nifi instance integration test.

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FlattenJson.java [hunk 1]
```diff
@@ -89,13 +92,32 @@
...
+            "Flattens every objects into a single level json");
+
+    public static final AllowableValue FLATTEN_MODE_KEEP_ARRAYS = new AllowableValue("keep arrays", "keep arrays",
+            "Flattens every objects and keep arrays format");
+
+    public static final AllowableValue FLATTEN_MODE_DOT_NOTATION = new AllowableValue("dot notation", "dot notation",
+            "Conforms to MongoDB dot notation to update also nested documents");
+
+    public static final PropertyDescriptor FLATTEN_MODE = new PropertyDescriptor.Builder()
+            .name("Flatten Mode")
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/FlattenJson.java]: Could you use both displayName() and name()?

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/groovy/org/apache/nifi/processors/standard/TestFlattenJson.groovy [hunk 2]
```diff
@@ -134,4 +134,44 @@ class TestFlattenJson {
...
+                first: [
+                        second: [
+                                third: [
+                                        "one", "two", "three", "four", "five"
+                                ]
+                        ]
+                ]
+        ]))
+
+        testRunner.setValidateExpressionUsage(true);
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/groovy/org/apache/nifi/processors/standard/TestFlattenJson.groovy]: not sure this line is required

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/groovy/org/apache/nifi/processors/standard/TestFlattenJson.groovy [hunk 3]
```diff
@@ -134,4 +134,44 @@ class TestFlattenJson {
...
```
Comment (pvillard31) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/groovy/org/apache/nifi/processors/standard/TestFlattenJson.groovy]: not sure this line is required

**Comment by yjhyjhyjh0** [comment]
Thanks for the review.
I've added display name as most processor does and removed unnecessary line at unit tests.

**Comment by MikeThomsen** [comment]
+1 LGTM.

**Pull request closed without merging**